### ISSUE TITLE
F.softmax(torch.tensor(2.), dim=0) doesn't make sense

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#46999]
## 🐛 Bug

F.softmax allows passing in a scalar tensor with dim=0. However, the scalar tensor has no dimensions, so asking to compute softmax over dimension 0 doesn't make sense.

## To Reproduce

```
>>> F.softmax(torch.tensor(2.), dim=0)
tensor(1.)
```

## Expected behavior

This should probably error out or we should update our documentation

## Environment
master

**Comment by zou3519** [comment]
FWIW numpy has the same behavior, so this is probably expected behavior:
```
x = np.array(2.)
x.sum(0)
```

**Comment by gchanan** [comment]
Yes, we allowed this because of numpy and also because 0-dim tensors came after reductions, so it would have been a compatibility nightmare.

**Comment by albanD** [comment]
So I guess we should just update the doc to mention this behavior and keep it as is?

**Comment by gchanan** [comment]
ya, that sounds good.
